src/migrations/1706698413211-full-text-search.ts

```diff
@@ -0,0 +1,64 @@
+      setweight(to_tsvector('german', name), 'A') || ' ' ||
+      setweight(to_tsvector('spanish', name), 'A') || ' ' ||
+      setweight(to_tsvector('english', COALESCE(description,'')), 'B') || ' ' ||
+      setweight(to_tsvector('french', COALESCE(description,'')), 'B') || ' ' ||
+      setweight(to_tsvector('italian', COALESCE(description,'')), 'B') || ' ' ||
+      setweight(to_tsvector('german', COALESCE(description,'')), 'B') || ' ' ||
+      setweight(to_tsvector('spanish', COALESCE(description,'')), 'B') || ' ' ||
+      setweight(to_tsvector('english', COALESCE(settings::jsonb->'tags','{}')), 'C') || ' ' ||
+      setweight(to_tsvector('french', COALESCE(settings::jsonb->'tags','{}')), 'C') || ' ' ||
+      setweight(to_tsvector('italian', COALESCE(settings::jsonb->'tags','{}')), 'C') || ' ' ||
+      setweight(to_tsvector('german', COALESCE(settings::jsonb->'tags','{}')), 'C') || ' ' ||
+      setweight(to_tsvector('spanish', COALESCE(settings::jsonb->'tags','{}')), 'C') || ' ' ||
+      setweight(to_tsvector('english', COALESCE(replace(extra, '\\u0000', '')::jsonb->'document'->>'content','{}')), 'D') || ' ' ||
+      setweight(to_tsvector('french', COALESCE(replace(extra, '\\u0000', '')::jsonb->'document'->>'content','{}')), 'D') || ' ' ||
+      setweight(to_tsvector('german', COALESCE(replace(extra, '\\u0000', '')::jsonb->'document'->>'content','{}')), 'D') || ' ' ||
+      setweight(to_tsvector('italian', COALESCE(replace(extra, '\\u0000', '')::jsonb->'document'->>'content','{}')), 'D') || ' ' ||
+      setweight(to_tsvector('spanish', COALESCE(replace(extra, '\\u0000', '')::jsonb->'document'->>'content','{}')), 'D') || ' ' ||
+      setweight(to_tsvector('english', COALESCE(replace(extra, '\\u0000', '')::jsonb->'file'->'content','{}')), 'D') || ' ' ||
+      setweight(to_tsvector('french', COALESCE(replace(extra, '\\u0000', '')::jsonb->'file'->'content','{}')), 'D') || ' ' ||
+      setweight(to_tsvector('italian', COALESCE(replace(extra, '\\u0000', '')::jsonb->'file'->'content','{}')), 'D') || ' ' ||
+      setweight(to_tsvector('german', COALESCE(replace(extra, '\\u0000', '')::jsonb->'file'->'content','{}')), 'D') || ' ' ||
+      setweight(to_tsvector('spanish', COALESCE(replace(extra, '\\u0000', '')::jsonb->'file'->'content','{}')), 'D') || ' ' ||
+      setweight(to_tsvector('english', COALESCE(replace(extra, '\\u0000', '')::jsonb->'s3File'->'content','{}')), 'D') || ' ' ||
+      setweight(to_tsvector('french', COALESCE(replace(extra, '\\u0000', '')::jsonb->'s3File'->'content','{}')), 'D') || ' ' ||
+      setweight(to_tsvector('italian', COALESCE(replace(extra, '\\u0000', '')::jsonb->'s3File'->'content','{}')), 'D') || ' ' ||
+      setweight(to_tsvector('german', COALESCE(replace(extra, '\\u0000', '')::jsonb->'s3File'->'content','{}')), 'D') || ' ' ||
+      setweight(to_tsvector('spanish', COALESCE(replace(extra, '\\u0000', '')::jsonb->'s3File'->'content','{}')), 'D')
```

Comment:
It's probably not a so good idea to add all the languages. But if we handle only the item language it means by default all existing item will have the english language, and only once you edit this setting you might get a good search. 
But maybe a good alternative would be to handle english and french by default (since they are our main languages) and all the other languages will "only" work if the setting is correct. (if the language is correctly set to french, finding  "chat" from "chats" works, but it won't if it's set to english)
```
vector(english) || vector(french) || vector(item language)  
```
That means we need to set a language! in a new column? What do you think?